application-glossary-machine-translation/application-glossary-machine-translation-api/src/main/java/org/xwiki/contrib/glossary/machineTranslation/internal/DefaultTranslationGlossaryManager.java

```diff
@@ -0,0 +1,142 @@
+
+import org.slf4j.Logger;
+import org.xwiki.component.annotation.Component;
+import org.xwiki.contrib.glossary.machineTranslation.TranslationGlossaryManager;
+import org.xwiki.contrib.translator.Translator;
+import org.xwiki.contrib.translator.TranslatorManager;
+import org.xwiki.contrib.translator.model.LocalePairs;
+import org.xwiki.contrib.translator.model.GlossaryUpdateEntry;
+import org.xwiki.model.reference.DocumentReference;
+import org.xwiki.model.reference.ObjectReference;
+import org.xwiki.query.Query;
+import org.xwiki.query.QueryException;
+import org.xwiki.query.QueryManager;
+
+import com.xpn.xwiki.XWikiContext;
+import com.xpn.xwiki.doc.XWikiDocument;
+import com.xpn.xwiki.objects.BaseObject;
+
+/**
+ * @version $Id$
+ */
+@Component
+@Singleton
+public class DefaultTranslationGlossaryManager implements TranslationGlossaryManager
+{
+    @Inject
+    private Logger logger;
+
+    @Inject
+    private Provider<XWikiContext> xwikiContextProvider;
+
+    @Inject
+    private TranslatorManager translatorManager;
+
+    @Inject
+    private QueryManager queryManager;
+
+    private Map<String, String> getLocalGlossaryEntries(Locale sourceLanguage, Locale targetLanguage)
+        throws QueryException
+    {
+        List<String[]> glossaryRawEntries = queryManager.createQuery("select sourceDoc.title, targetDoc.title "
+                    + "from XWikiDocument sourceDoc, XWikiDocument targetDoc, BaseObject glossaryObj "
+                    + "where glossaryObj.name = sourceDoc.fullName "
+                    + "and glossaryObj.name = targetDoc.fullName "
+                    + "and glossaryObj.className = 'Glossary.Code.GlossaryClass' "
+                    + "and sourceDoc.space = 'Glossary' "
+                    + "and targetDoc.space = 'Glossary' "
+                    + "and ((sourceDoc.defaultLanguage = :sourceLanguage and sourceDoc.language = '') "
+                    + "     or sourceDoc.language = :sourceLanguage) "
+                    + "and ((targetDoc.defaultLanguage = :targetLanguage and targetDoc.language = '') "
+                    + "     or targetDoc.language = :targetLanguage)",
+                Query.HQL).bindValue("sourceLanguage", sourceLanguage.toString())
+            .bindValue("targetLanguage", targetLanguage.toString()).execute();
+
+        Map<String, String> glossaryEntries = new HashMap<>();
+        for (Object[] entry : glossaryRawEntries) {
+            String trimmedKey = ((String) entry[0]).trim();
+            String trimmedValue = ((String) entry[1]).trim();
+            glossaryEntries.put(trimmedKey, trimmedValue);
+        }
+        return glossaryEntries;
+    }
+
+    @Override
+    public void synchronizeGlossaries()
+    {
+        Translator translator = translatorManager.getTranslator();
+        XWikiContext context = xwikiContextProvider.get();
+
+        try {
+            List<LocalePairs> translatorSupportedGlossaries = translator.getGlossaryLocalePairs();
```

Comment:
Since we are retrieving locale pairs here, not glossaries, maybe another variable name would be more appropriate?

Reply:
Fixed by 67756a1